packages/io/json-deserializer/index.js

```diff
@@ -1,110 +1,100 @@
 /*
 ## License
 
-Copyright (c) 2016 Z3 Development https://github.com/z3dev
+Copyright (c) JSCAD Organization https://github.com/jscad
 
 All code released under MIT license
-
-History:
-  2016/10/15: 0.5.2: initial version
-
-Notes:
-1) All functions extend other objects in order to maintain namespaces.
 */
 
-// import { CSG } from '@jscad/csg'
-const {CSG, CAG} = require('@jscad/csg')
-
 // //////////////////////////////////////////
 //
 // JSON (JavaScript Object Notation) is a lightweight data-interchange format
-// See http://json.org/
+// See https://www.json.org
 //
 // //////////////////////////////////////////
 
-function toSourceCSGVertex (ver) {
-  return 'new CSG.Vertex(new CSG.Vector3D(' + ver._x + ',' + ver._y + ',' + ver._z + '))'
-}
+const { geometry } = require('@jscad/modeling')
 
-// convert the give CSG object to JSCAD source
-function toSourceCSG (csg) {
-  var code = '  var polygons = [];\n'
-  csg.polygons.map(function (p) {
-    code += '  poly = new CSG.Polygon([\n'
-    for (var i = 0; i < p.vertices.length; i++) {
-      code += '                         ' + toSourceCSGVertex(p.vertices[i].pos) + ',\n'
-    }
-    code += '                         ])'
-    if (p.shared && p.shared.color && p.shared.color.length) {
-      code += '.setColor(' + JSON.stringify(p.shared.color) + ');\n'
-    } else {
-      code += ';\n'
-    }
-    code += '  polygons.push(poly);\n'
-  })
-  code += '  return CSG.fromPolygons(polygons);\n'
-  return code
-};
-
-function toSourceCAGVertex (ver) {
-  return 'new CAG.Vertex(new CSG.Vector2D(' + ver.pos._x + ',' + ver.pos._y + '))'
-};
-function toSourceSide (side) {
-  return 'new CAG.Side(' + toSourceCAGVertex(side.vertex0) + ',' + toSourceCAGVertex(side.vertex1) + ')'
-};
-
-// convert the give CAG object to JSCAD source
-function toSourceCAG (cag) {
-  var code = '  var sides = [];\n'
-  cag.sides.map(function (s) {
-    code += '  sides.push(' + toSourceSide(s) + ');\n'
-  })
-  code += '  return CAG.fromSides(sides);\n'
-  return code
-}
+const { flatten, toArray } = require('@jscad/array-utils')
 
-// convert an anonymous CSG/CAG object to JSCAD source
-function toSource (obj) {
-  if (obj.type && obj.type === 'csg') {
-    var csg = CSG.fromObject(obj)
-    return toSourceCSG(csg)
-  }
-  if (obj.type && obj.type === 'cag') {
-    var cag = CAG.fromObject(obj)
-    return toSourceCAG(cag)
+const version = require('./package.json').version
+
+/**
+ * Deserialize the given JSON notation (string) into either a script or an array of geometry.
```

Comment:
so all of this is very cool !
BUT , we still need some specific code to deal with TypedArrays
Good news is, the vtree V2 code has that, bad news is , I am not done with it :D